pipeline/relation_embedder/relation_embedder/src/main/scala/weco/pipeline/relation_embedder/CollectionPathSorter.scala

```diff
@@ -1,80 +0,0 @@
-package weco.pipeline.relation_embedder
-
-import scala.util.Try
-import scala.annotation.tailrec
-
-import weco.catalogue.internal_model.work.Work
-import weco.catalogue.internal_model.work.WorkState.Identified
-
-object CollectionPathSorter {
-
-  def sortWorks(works: List[Work[Identified]]): List[Work[Identified]] =
-    works.sortBy(tokenizePath)
-
-  def sortPaths(paths: List[String]): List[String] =
-    paths.sortBy(tokenizePath)
-
-  type TokenizedPath = List[PathToken]
-  type PathToken = List[PathTokenPart]
-  type PathTokenPart = Either[Int, String]
-
-  private def tokenizePath(path: String): TokenizedPath =
-    path.split("/").toList.map { str =>
-      """\d+|\D+""".r
-        .findAllIn(str)
-        .toList
-        .map { token =>
-          Try(token.toInt).map(Left(_)).getOrElse(Right(token))
-        }
-    }
-
-  private def tokenizePath(work: Work[Identified]): Option[TokenizedPath] =
-    work.data.collectionPath
-      .map { collectionPath =>
-        tokenizePath(collectionPath.path)
-      }
-
-  implicit val tokenizedPathOrdering: Ordering[TokenizedPath] =
-    new Ordering[TokenizedPath] {
-      @tailrec
-      override def compare(a: TokenizedPath, b: TokenizedPath): Int =
-        (a, b) match {
-          case (Nil, Nil) => 0
-          case (Nil, _)   => -1
-          case (_, Nil)   => 1
-          case (xHead :: xTail, yHead :: yTail) =>
-            if (xHead == yHead)
-              compare(xTail, yTail)
-            else
-              pathTokenOrdering.compare(xHead, yHead)
-        }
-    }
-
-  implicit val pathTokenOrdering: Ordering[PathToken] =
-    new Ordering[PathToken] {
-      @tailrec
-      override def compare(a: PathToken, b: PathToken): Int =
-        (a, b) match {
-          case (Nil, Nil) => 0
-          case (Nil, _)   => 1
-          case (_, Nil)   => -1
```

Comment:
These two lines are the cause of the behaviour Victoria is seeing – they used to cause longer parts to sort higher. Notice the difference between this and tokenizedPathOrdering:

```scala
          case (Nil, _)   => -1     // pathTokenOrdering
          case (_, Nil)   => 1

          case (Nil, _)   => 1      // tokenizedPathOrdering
          case (_, Nil)   => -1
```

e.g. consider sorting `A/B/1` and `A/B/1b`.

The old code would sort `1b` above `1` because it was longer. When I created the common `shorterPathsWinOrdering` behaviour, I flipped this behaviour – so we now get the desired behaviour "for free".

(Also this is why you need tests! This is a subtle behaviour change that I wouldn't have noticed if I hadn't been looking for it, and wasn't caught by the tests.)